epochStart/shardchain/trigger.go

```diff
@@ -741,7 +773,20 @@ func (t *trigger) getHeaderWithNonceAndPrevHash(nonce uint64, prevHash []byte) (
 		return nil, marshal.ErrUnmarshallingBadSize
 	}
 
-	return t.getHeaderWithNonceAndHash(nonce, b.Data[0])
+	metaHdrs := make([]*block.MetaBlock, 0)
```

Comment:
I think that this new part of code is useless as always the return of the dataHdr, err := t.metaNonceHdrStorage.Get(nonceToByteSlice)
will have maximum one record. As this storage will overwrite the latest nonce committed. Also the batch used above is useless

Reply:
it is useless to search in storage at that moment. the finality attesting metaheader should be in the pool or in the local cache.

Reply:
reverted as lower round search was done in previous iterations.